packages/@uppy/core/src/Restricter.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable max-classes-per-file, class-methods-use-this */
+/* global AggregateError */
+const prettierBytes = require('@transloadit/prettier-bytes')
+const match = require('mime-match')
+
+const defaultOptions = {
+  maxFileSize: null,
+  minFileSize: null,
+  maxTotalFileSize: null,
+  maxNumberOfFiles: null,
+  minNumberOfFiles: null,
+  allowedFileTypes: null,
+  requiredMetaFields: [],
+}
+
+class RestrictionError extends Error {
+  constructor (...args) {
+    super(...args)
+    this.isRestriction = true
+  }
+}
+
+if (typeof AggregateError === 'undefined') {
+  // eslint-disable-next-line no-global-assign
+  globalThis.AggregateError = class AggregateError extends Error {
+    constructor (errors, message) {
+      super(message)
+      this.errors = errors
+    }
+  }
+}
+
+class AggregateRestrictionError extends AggregateError {
+  constructor (...args) {
+    super(...args)
+    this.isRestriction = true
+  }
+}
+
+class Restricter {
```

Comment:
would it better to just export the functions instead of wrapping them in a class? I don't see the benefit of using a class here. then we could also remove `eslint-disable class-methods-use-this `

Reply:
I'm inclined to keep it a class to do this: 
> There are repetitive things being injected, such as `i18n` and `opts`. Perhaps better to add these in the constructor.

Reply:
Alright, with contructor arguments it makes sense to use a class. However this class now has two ways of passing dependencies: one is through the constructor, and the other is through `ctx`. Should we pass the ctx properties also in through the constructor to make it consistent?